openapi/client.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ARGC=$#
+
+if [ $# -ne 3 ]; then
+    echo "Usage:"
+    echo "  $(basename ${0}) OUTPUT_DIR SETTING_FILE_PATH CLIENT_LANGUAGE"
+    echo "    Setting file should define KUBERNETES_BRANCH, CLIENT_VERSION, and PACKAGE_NAME"
+    echo "    Setting file can define an optional USERNAME if you're working on a fork"
+    echo "    Setting file can define an optional REPOSITORY if you're working on a ecosystem project"
+    exit 1
+fi
+
+OUTPUT_DIR=$1
+SETTING_FILE=$2
+CLIENT_LANGUAGE=${3,,}
+
+case $CLIENT_LANGUAGE in
+
+  c)
+    OPENAPI_GENERATOR_COMMIT="${OPENAPI_GENERATOR_COMMIT:-master}"
+    CLEANUP_DIRS=(pkg)
+    ;;
+
+  csharp)
+    CLEANUP_DIRS=(docs src target gradle)
+    ;;
+
+  go)
+    OPENAPI_GENERATOR_COMMIT="${OPENAPI_GENERATOR_COMMIT:-v3.3.4}"
+    CLEANUP_DIRS=(pkg)
+    ;;
+
+  haskell)
+    OPENAPI_GENERATOR_COMMIT="${OPENAPI_GENERATOR_COMMIT:-a979fd8e13c86431831b0c769ba7b484e744afa5}"
+    CLEANUP_DIRS=(lib/Kubernetes/OpenAPI/API)
+    ;;
+
+  java)
```

Comment:
```suggestion
  java)
    OPENAPI_GENERATOR_COMMIT="${OPENAPI_GENERATOR_COMMIT:-v4.2.0}" \
```

java generation has a strict bound on the generator version